controllers/dscinitialization/servicemesh_setup.go

```diff
@@ -130,3 +142,47 @@ func configureServiceMeshFeatures() feature.FeaturesProvider {
 		return nil
 	}
 }
+
+func IsDefaultAudiences(specAudiences *[]string) bool {
+	return specAudiences == nil || reflect.DeepEqual(*specAudiences, defaultAudiences)
+}
+
+// DefinedAudiencesOrDefault returns the default audiences if the provided audiences are default, otherwise it returns the provided audiences.
+func DefinedAudiencesOrDefault(specAudiences *[]string) []string {
+	if IsDefaultAudiences(specAudiences) {
+		return fetchClusterAudiences()
+	}
+	return *specAudiences
+}
+
+func fetchClusterAudiences() []string {
+	restCfg, err := config.GetConfig()
+	if err != nil {
+		smSetupLog.Error(err, "Error getting config, using default audiences")
+		return defaultAudiences
+	}
+
+	tokenReview := &authentication.TokenReview{
+		Spec: authentication.TokenReviewSpec{
+			Token: restCfg.BearerToken,
+		},
+	}
+
+	tokenReviewClient, err := client.New(restCfg, client.Options{})
```

Comment:
Maybe it's worth making client (and log) part of the func params? We have `r.Client` and `r.Log`  available as part of reconciler (caller) struct.

Reply:
done! 
e38a8ad